prog/update_vm_ipv6.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require_relative "../lib/util"
+
+class Prog::UpdateVmIpv6 < Prog::Base
+  subject_is :vm
+
+  def vm_host
+    @vm_host ||= VmHost[vm.vm_host_id]
+  end
+
+  label def start
+    vm_host.sshable.cmd("sudo systemctl stop #{vm.inhost_name}.service")
+    vm_host.sshable.cmd("sudo systemctl stop #{vm.inhost_name}-dnsmasq.service")
+    vm_host.sshable.cmd("sudo ip netns del #{vm.inhost_name}")
+    hop_rewrite_persisted
+  end
+
+  label def rewrite_persisted
+    vm.update(
+      ephemeral_net6: vm_host.ip6_random_vm_network.to_s
+    )
+
+    write_params_json
+    vm_host.sshable.cmd("sudo host/bin/setup-vm reassign-ip6 #{vm.inhost_name}", stdin: JSON.generate({
+      storage: vm.vm_storage_volumes.filter_map { |s|
+        if !s.key_encryption_key_1.nil?
+          [s.device_id, s.key_encryption_key_1.secret_key_material_hash]
+        end
+      }.to_h
+    }))
+    hop_start_vm
+  end
+
+  label def start_vm
+    nic = vm.nics.first
+    addr = nic.private_subnet.net4.nth(1).to_s + nic.private_subnet.net4.netmask.to_s
+
+    vm_host.sshable.cmd("sudo ip -n #{vm.inhost_name.shellescape} addr replace #{addr} dev #{nic.ubid_to_tap_name}")
+    vm_host.sshable.cmd("sudo systemctl start #{vm.inhost_name}.service")
+    vm.incr_update_firewall_rules
+    pop "VM #{vm.name} updated"
+  end
+
+  def write_params_json
+    vm_host.sshable.cmd("sudo rm /vm/#{vm.inhost_name}/prep.json")
+    topo = vm.cloud_hypervisor_cpu_topology
+    params_json = JSON.pretty_generate({
+      "vm_name" => vm.name,
+      "public_ipv6" => vm.ephemeral_net6.to_s,
+      "public_ipv4" => vm.ip4.to_s || "",
+      "local_ipv4" => vm.local_vetho_ip&.to_s&.shellescape,
+      "unix_user" => vm.unix_user,
+      "ssh_public_key" => vm.public_key,
+      "nics" => vm.nics.map { |nic| [nic.private_ipv6.to_s, nic.private_ipv4.to_s, nic.ubid_to_tap_name, nic.mac] },
+      "boot_image" => vm.boot_image,
+      "max_vcpus" => topo.max_vcpus,
+      "cpu_topology" => topo.to_s,
+      "mem_gib" => vm.mem_gib,
+      "ndp_needed" => vm_host.ndp_needed,
+      "storage_volumes" => storage_volumes,
+      "swap_size_bytes" => nil,
+      "pci_devices" => vm.pci_devices.map { [_1.slot, _1.iommu_group] }
+    })
```

Comment:
I was actually thinking this would either be used a single time by an operator and then get cleaned up, or as an example script. If you think we can keep it around and productionize, I can work on it more.